unity/Assets/Maroon/GlobalEntities/WebGLUrlParameterReader/Scripts/WebGLUrlParameterReader.cs

```diff
@@ -1,12 +1,37 @@
 ﻿using UnityEngine;
 using System.Runtime.InteropServices;
+using System.Collections.Generic;
+using System;
+using Newtonsoft.Json;
 
 namespace Maroon
 {
-    public class WebGLUrlParameterReader : MonoBehaviour
+    public enum WebGlUrlParameter
+    {
+        LoadScene,
+        Config,
+    }
+    public class WebGlUrlParameterReader : MonoBehaviour
     {
         // JS implementation at Plugins/WebGLUrlParameterReader.jslib
         [DllImport("__Internal")]
-        public static extern string GetUrlParameter(string urlParameterName);
+        private static extern string _getUrlParameter(string urlParameterName);
+        [DllImport("__Internal")]
+        private static extern IntPtr _getAllUrlParameters();
+
+        public static string GetUrlParameter(WebGlUrlParameter urlParameter)
+        {
+            Debug.LogWarning("GetUrlParameter is deprecated due to case-sensitivity. Use 'BootstrappingManager.UrlParameters' instead.");
+            return GetAllUrlParameters().TryGetValue(urlParameter, out string value) ? value : null;
+        }
+
+        public static Dictionary<WebGlUrlParameter, string> GetAllUrlParameters()
+        {
+            // rawJson = {"LoadScene":"Optics","Config":"Default"}
+            var rawJson = Marshal.PtrToStringAnsi(_getAllUrlParameters());
+            var values = JsonConvert.DeserializeObject<Dictionary<WebGlUrlParameter, string>>(rawJson);
```

Comment:
I didn't look into this tbh, but is this safe from a security point of view? i.e. could a malicious user send an innocent person a link to a Maroon website which has like `?LoadScene=Optics&config=maliciousCodeSomethingIdk` at the end?